test/library/packages/UnitTest/test_locales.chpl

```diff
@@ -49,4 +49,29 @@ proc s9(test: Test) throws {
   test.addNumLocales(8,16);
 }
 
-UnitTest.runTest(s1,s2,s3,s4,s5,s6,s7,s8,s9);
+// this test halts
+proc s10(test: Test) throws {
+  var a: [1..0] int;
+  writeln(a[0]);
+}
+
+// this depend on test s9 which depends on s10 which halts
+proc s11(test: Test) throws {
+  test.dependsOn(s14,s1);
+}
+
+// this depends on a test that pass
+proc s12(test: Test) throws {
+  test.dependsOn(s2);
+}
+
+// this depends on a test which fails
+proc s13(test: Test) throws {
+  test.dependsOn(s8);
+}
+
+// depends on a test that halts
+proc s14(test: Test) throws {
+  test.dependsOn(s10);
+}
```

Comment:
One test case that just occurred to me:
- Test B depends on Test A
- Test C depends on Test B and Test A (and then check swapping their order as well)

Basically, Test A shows up in the dependency list twice, but isn't actually part of a circular dependency.

Reply:
Thanks, I forgot about this situation. I have made the changes and now I am removing the already ran tests so that we can tackle these.

Reply:
Thanks, I forgot about this situation. I have made the changes and now I am removing the already ran tests so that we can tackle these.

Reply:
Added test for this.